src/pytesmo/validation_framework/adapters.py

```diff
@@ -124,6 +184,11 @@ def read(self, *args, **kwargs):
         data = super(MaskingAdapter, self).read(*args, **kwargs)
         return self.__mask(data)
 
+    def _adapt_custom(self, *args, **kwargs):
+        data = super()._adapt_custom(*args, **kwargs)
+        return self.__mask(data)
+
+
 class SelfMaskingAdapter(BasicAdapter):
     """
     Transform the given (reader) class to return a dataset that is masked based
```

Comment:
What is actually the difference between `MaskingAdapter` and `SelfMaskingAdapter`? They look very redundant to me.

Also, I think those should only override the `_adapt` method of the parent class, the other methods can be removed. Something like:
```
def _adapt(self, data):
    mask = self.operator(data[self.column_name], self.threshold)
    return data[mask]
```

Reply:
good question, looks like one picks a single column to mask, the other masks the whole data frame...anyway I dont see the point of `MaskingAdapter` (and we dont use that one anywhere as far as I am aware), I would mark it as deprecated and remove it next time.

I think `__adapt` should work (otherwise the parent/child class get mixed up)

Reply:
> I think __adapt should work (otherwise the parent/child methods get mixed up)

The parent method can still be called with `super()._adapt`, for example like this:
````
def _adapt(self, data):
    data = super()._adapt(data)
    mask = self.operator(data[self.column_name], self.threshold)
    return data[mask]
```
Do we actually need to distinguish between the methods of parent and child class?

Reply:
We just have to make sure that all of them are called ;) and if you adapt and already adapted reader as well etc.

Reply:
See https://github.com/TUW-GEO/pytesmo/pull/245/commits/602b5b6281a82d151f38a9955ddab67acb877dac
I guess this is what you meant? I like that solution, the only "problem" is that the adaptation function of the BasicAdapter is called multiple time when stacking multiple adapters. This is not really a problem because in those cases nothing happens then. Just a few unnecessary attribute checks.